crypto/mem.c

```diff
@@ -31,6 +31,14 @@ static void (*free_impl)(void *, const char *, int)
     = CRYPTO_free;
 
 #ifndef OPENSSL_NO_CRYPTO_MDEBUG
+static int malloc_count;
+static int realloc_count;
+static int free_count;
+static int dummy;
+
+# define INCREMENT(x) CRYPTO_atomic_add(&x, 1, &dummy, malloc_lock)
+# define GET(ret, val) CRYPTO_atomic_add(&val, 0, ret, malloc_lock)
```

Comment:
On side note I find malloc_lock name *extremely* confusing, memdbg_lock would be better...

Reply:
Agreed.

Reply:
I can change it if you want.  There are to malloc-lock and malloc-lock-long.  additional commit tomorrow